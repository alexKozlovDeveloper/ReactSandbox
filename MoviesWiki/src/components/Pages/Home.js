import React, { Component } from "react";

import ResultsBody from "../ResultsBody";

import Header from "../Header";
import Footer from "../Footer";

import styles from "../../styles/Home.css"

const Home = ({ resultsBodyConfig, headerConfig, match, history }) => {
    return <div className={styles.home}>
        <Header config={headerConfig}/>
        <ResultsBody config={resultsBodyConfig}></ResultsBody>
        <Footer />
    </div>;
}

export default Home;