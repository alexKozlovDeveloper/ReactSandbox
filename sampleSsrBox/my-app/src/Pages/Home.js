import React from "react";
import { connect } from 'react-redux'
import { updateTitle } from '../actions'

const Home = ({ title }) => {
    return <div>This ]Home[ page [{title}]</div>
}

function mapStateToProps(state) {
    const { title } = state.TitleReducer;
    console.log(state, "home");
    return {
        title
    };
}

export default connect(
    mapStateToProps
)(Home)
