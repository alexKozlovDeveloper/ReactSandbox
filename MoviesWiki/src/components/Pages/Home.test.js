import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { createStore } from 'redux';
import { Provider } from 'react-redux';

import { BrowserRouter as Router } from 'react-router-dom';
import rootReducer from '../../reducers';

import Home from './Home';

configure({ adapter: new Adapter() });

test('Home rendering test', () => {
  const store = createStore(rootReducer);

  const resultsBodyConfig = {
    itemsPerRow: 4,
    resultSortConfig: {
      title: 'Sort by',
      options: [{
        title: 'Release date',
        value: 'release_date',
      }, {
        title: 'Rating',
        value: 'vote_average',
      }],
    },
    resultsCountConfig: {
      title: 'movies found',
    },
  };
  const headerConfig = {
    title: 'Find Your Movies',
    searchConfig: {
      placeHolder: 'The Godfather',
      buttonText: 'Search',
      searchFilter: {
        title: 'Search by',
        options: [{
          title: 'Title',
          value: 'title',
        }, {
          title: 'Genre',
          value: 'genres',
        }],
      },
    },
  };

  const {
    Response, Request, Headers, fetch,
  } = require('whatwg-fetch');
  global.Response = Response;
  global.Request = Request;
  global.Headers = Headers;
  global.fetch = fetch;

  const component = mount(<Router><Provider store={store}><Home resultsBodyConfig={resultsBodyConfig} headerConfig={headerConfig}/></Provider></Router>);
  expect(shallowToJson(component)).toMatchSnapshot();
});
