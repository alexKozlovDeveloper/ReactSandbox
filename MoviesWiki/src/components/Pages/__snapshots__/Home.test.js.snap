// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home rendering test 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Home
        headerConfig={
          Object {
            "searchConfig": Object {
              "buttonText": "Search",
              "placeHolder": "The Godfather",
              "searchFilter": Object {
                "options": Array [
                  Object {
                    "title": "Title",
                    "value": "title",
                  },
                  Object {
                    "title": "Genre",
                    "value": "genres",
                  },
                ],
                "title": "Search by",
              },
            },
            "title": "Find Your Movies",
          }
        }
        resultsBodyConfig={
          Object {
            "itemsPerRow": 4,
            "resultSortConfig": Object {
              "options": Array [
                Object {
                  "title": "Release date",
                  "value": "release_date",
                },
                Object {
                  "title": "Rating",
                  "value": "vote_average",
                },
              ],
              "title": "Sort by",
            },
            "resultsCountConfig": Object {
              "title": "movies found",
            },
          }
        }
      >
        <div
          className="home"
        >
          <Header
            config={
              Object {
                "searchConfig": Object {
                  "buttonText": "Search",
                  "placeHolder": "The Godfather",
                  "searchFilter": Object {
                    "options": Array [
                      Object {
                        "title": "Title",
                        "value": "title",
                      },
                      Object {
                        "title": "Genre",
                        "value": "genres",
                      },
                    ],
                    "title": "Search by",
                  },
                },
                "title": "Find Your Movies",
              }
            }
          >
            <div
              className="header"
            >
              <div
                className="container"
              >
                <div
                  className="headertitle"
                >
                  Find Your Movies
                </div>
                <withRouter(Connect(Search))
                  config={
                    Object {
                      "buttonText": "Search",
                      "placeHolder": "The Godfather",
                      "searchFilter": Object {
                        "options": Array [
                          Object {
                            "title": "Title",
                            "value": "title",
                          },
                          Object {
                            "title": "Genre",
                            "value": "genres",
                          },
                        ],
                        "title": "Search by",
                      },
                    }
                  }
                >
                  <Connect(Search)
                    config={
                      Object {
                        "buttonText": "Search",
                        "placeHolder": "The Godfather",
                        "searchFilter": Object {
                          "options": Array [
                            Object {
                              "title": "Title",
                              "value": "title",
                            },
                            Object {
                              "title": "Genre",
                              "value": "genres",
                            },
                          ],
                          "title": "Search by",
                        },
                      }
                    }
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <Search
                      config={
                        Object {
                          "buttonText": "Search",
                          "placeHolder": "The Godfather",
                          "searchFilter": Object {
                            "options": Array [
                              Object {
                                "title": "Title",
                                "value": "title",
                              },
                              Object {
                                "title": "Genre",
                                "value": "genres",
                              },
                            ],
                            "title": "Search by",
                          },
                        }
                      }
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      isLoaded={false}
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      movies={Array []}
                      searchBy="title"
                      searchText=""
                      updateMoviesBySeachFunc={[Function]}
                      updateSearchByFunc={[Function]}
                      updateSearchTextFunc={[Function]}
                    >
                      <div
                        className="search"
                      >
                        <div>
                          <SearchField
                            placeHolder="The Godfather"
                            searchText=""
                            submitFunc={[Function]}
                            updateFunc={[Function]}
                          >
                            <input
                              className="searchfield"
                              defaultValue=""
                              onInput={[Function]}
                              onKeyDown={[Function]}
                              placeholder="The Godfather"
                              type="text"
                            />
                          </SearchField>
                          <SearchButton
                            buttonText="Search"
                            updateFunc={[Function]}
                          >
                            <button
                              className="button"
                              onClick={[Function]}
                            >
                              Search
                            </button>
                          </SearchButton>
                        </div>
                        <CustomOptionList
                          config={
                            Object {
                              "options": Array [
                                Object {
                                  "title": "Title",
                                  "value": "title",
                                },
                                Object {
                                  "title": "Genre",
                                  "value": "genres",
                                },
                              ],
                              "title": "Search by",
                            }
                          }
                          selectedIndex="0"
                          updateFunc={[Function]}
                        >
                          <div
                            className="container"
                          >
                            <div
                              className="title"
                            >
                              Search by
                            </div>
                            <div
                              className="item isactive"
                              key="0"
                              name="option0"
                              onClick={[Function]}
                              value="title"
                            >
                              Title
                            </div>
                            <div
                              className="item"
                              key="1"
                              name="option1"
                              onClick={[Function]}
                              value="genres"
                            >
                              Genre
                            </div>
                          </div>
                        </CustomOptionList>
                      </div>
                    </Search>
                  </Connect(Search)>
                </withRouter(Connect(Search))>
              </div>
            </div>
          </Header>
          <Connect(ResultsBody)
            config={
              Object {
                "itemsPerRow": 4,
                "resultSortConfig": Object {
                  "options": Array [
                    Object {
                      "title": "Release date",
                      "value": "release_date",
                    },
                    Object {
                      "title": "Rating",
                      "value": "vote_average",
                    },
                  ],
                  "title": "Sort by",
                },
                "resultsCountConfig": Object {
                  "title": "movies found",
                },
              }
            }
          >
            <ResultsBody
              config={
                Object {
                  "itemsPerRow": 4,
                  "resultSortConfig": Object {
                    "options": Array [
                      Object {
                        "title": "Release date",
                        "value": "release_date",
                      },
                      Object {
                        "title": "Rating",
                        "value": "vote_average",
                      },
                    ],
                    "title": "Sort by",
                  },
                  "resultsCountConfig": Object {
                    "title": "movies found",
                  },
                }
              }
              isLoaded={false}
              movies={Array []}
              sortBy="vote_average"
              updateMoviesFunc={[Function]}
              updateSelectedItemFunc={[Function]}
              updateSortByFunc={[Function]}
            >
              <div
                className="body"
              >
                <Spinner>
                  <div
                    className="spinnercontainer"
                  >
                    <div
                      className="spinner"
                    >
                      <div />
                      <div />
                      <div />
                    </div>
                  </div>
                </Spinner>
              </div>
            </ResultsBody>
          </Connect(ResultsBody)>
          <Footer>
            <div
              className="footer"
            >
              <div
                className="container"
              >
                Copyright © 2019 AkTest
              </div>
            </div>
          </Footer>
        </div>
      </Home>
    </Provider>
  </Router>
</BrowserRouter>
`;
